packages/dataviews/src/utils.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Helper util to sort items by text fields, when sorting is done client side.
+ *
+ * @param {Object}   params            Function params.
+ * @param {Object[]} params.items      Array of items to sort.
+ * @param {Object}   params.view       Current view object.
+ * @param {Object[]} params.fields     Array of available fields.
+ * @param {string[]} params.textFields Array of the field ids to sort.
+ *
+ * @return {Object[]} Sorted items.
+ */
+export const sortByTextFields = ( { items, view, fields, textFields } ) => {
```

Comment:
I don't understand why we need the text fields argument, isn't the sorting config part of the "view" object

Reply:
For me it seems the type of sorting is related to the field, but if we have an API prop in fields it will only be used for client side(that's what Tanstack does too - sortingFn). So I don't think we need to add such an API like that for now.. Did you have something different in your mind?

Reply:
The ideal scenario for me is that the component end up with a single hook abstracting all of this at some point:

```
// probably a bad name
const { isLoading, data, paginationInfo } = useLocalRequest( allData, view, fields )
```

separating textFields means that we have to pass an extra argument to such hooks. Isn't the "type" of the field sufficient here?

Reply:
Hm.. I'm not sure yet if we could use just the `type`. For example in templates `author` the type is `enumeration` and is used for filters and it could be used for inline editing. But for sorting we just want to do it as a `text` field.

Reply:
isn't "enumeration" a special text field?

Reply:
Maybe yes, but I'm not too eager to abstract right now because as I said, I think we will be able to make more informed decisions when we evolve the other APIs too.. While not pretty right now, we just pass an array and will be much simpler to change in the near future.